php-checks/src/main/java/org/sonar/php/checks/wordpress/WordPressAutoUpdateCheck.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.php.checks.wordpress;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Predicate;
+import org.sonar.check.Rule;
+import org.sonar.php.checks.utils.CheckUtils;
+import org.sonar.plugins.php.api.tree.Tree;
+import org.sonar.plugins.php.api.tree.expression.ExpressionTree;
+import org.sonar.plugins.php.api.tree.expression.FunctionCallTree;
+import org.sonar.plugins.php.api.tree.expression.LiteralTree;
+
+@Rule(key = "S6343")
+public class WordPressAutoUpdateCheck extends WordPressConfigVisitor {
+
+  private static final String MESSAGE = "Make sure deactivating automatic updates is intended.";
+
+  @Override
+  protected Set<String> configsToVisit() {
+    return new HashSet<>(Arrays.asList("AUTOMATIC_UPDATER_DISABLED", "WP_AUTO_UPDATE_CORE", "DISALLOW_FILE_MODS"));
+  }
+
+  @Override
+  void visitConfigDeclaration(FunctionCallTree config) {
+    configKey(config)
+      .filter(c -> c.is(Tree.Kind.REGULAR_STRING_LITERAL))
+      .map(c -> CheckUtils.trimQuotes((LiteralTree) c))
```

Comment:
I would like to create a function Optional<String> configKeyAsString(FunctionCallTree) which will return a equate trimmed optional String. We can do it in another PR. Nevertheless, it would also suit your previous check.

Reply:
c3fdd78